motion_tracker/utils/image_generator.py

```diff
@@ -0,0 +1,193 @@
+        output_height: int
+            Height in pixels of output images for start and ending image
+
+    """
+
+    generators_to_draw_from = [imagenet_generator(crops_per_image), 
+                               alov_generator(crops_per_image)]
+
+    empty_img_accumulator = np.zeros([0, output_width, output_height, 3]).astype('uint8')
+    empty_box_accumulator = np.zeros([0, 4])
+
+    while True:
+        my_output = {'start_img': empty_img_accumulator,
+                     'start_box': empty_box_accumulator,
+                     'end_img': empty_img_accumulator,
+                     'end_box': empty_box_accumulator}
+
+        # Check count of images already in batch is less than batch_size.
+        while my_output['start_img'].shape[0] < batch_size:
+            for img_source in generators_to_draw_from:
+                subsource_output = next(img_source)
+                for field in my_output:
+                    # Add outputs from subsource
+                    my_output[field] = np.concatenate([my_output[field],
+                                                       subsource_output[field]])
+            
+        # Limit output arrays from being larger than batch_size
+        for field in my_output:
+            my_output[field] = my_output[field][:batch_size]
+        yield my_output
+
+def set_array_dims(img, box):
+    ''' Adds an additional axis of length 1 to start of imgs and boxes.
+
+        These dimensionality changes are necessary for merging to accumulators
+        that have observations in the first axis
+    '''
+
+    out_img = img[np.newaxis].astype('uint8')
+    out_box = box.as_array()[np.newaxis]
+    return out_img, out_box
+
+def imagenet_generator(crops_per_image=10, batch_size=50,
+                       output_width = 256, output_height = 256):
+    """Generator yielding dictionary of image and bounding boxes.
+
+    Args:
+    ----
+        crops_per_image: int
+            Number of random crops of the current frame to take.
+            Corresponds to both k3 and k4 in the paper, with a value of 10.
+        batch_size: int
+            Number of items in list to be returned.
+        output_width: int
+            Width in pixels of output images for start and ending image
+        output_height: int
+            Height in pixels of output images for start and ending image
+
+    """
+
+    raw_image_dir = 'data/imagenet/images/'
+    img_metadata = pd.read_csv('work/imagenet/parsed_bb2.csv')
+
+    empty_img_accumulator = np.zeros([0, output_width, output_height, 3]).astype('uint8')
+    empty_box_accumulator = np.zeros([0, 4])
+    while True:
+        my_output = {'start_img': empty_img_accumulator,
+                     'start_box': empty_box_accumulator,
+                     'end_img': empty_img_accumulator,
+                     'end_box': empty_box_accumulator}
+        while my_output['start_img'].shape[0] < batch_size:
```

Comment:
I think this is the only remaining issue here. We need the imagenet generator and the alov generator to generate 10 image pairs each, rather than the batch size. If we generate the batch size, then I think the way that the master generator is built, we'll end up training on only alov or imagenet for each batch, and not both. I understand we want to keep this in a similar format as the master generator, so I was hesitant to switch the `< batch_size` part here. It'd be easy if we wanted to to have some bool passed in that notes whether the generator is being used alone (and this line should use batch size) or as part of the master generator (and this line should be using crops_per_image).